Makefile

```diff
@@ -126,10 +129,16 @@ clean:
 	rm -rf ./build
 	rm -rf emulator/obj_dir
 	rm -f emulator/*.v
-	rm -f $(LIB_DIR)/include/flexpret_config.h $(LIB_DIR)/linker/flexpret_config.ld
+	rm -f $(LIB_DIR)/include/flexpret_hwconfig.h $(LIB_DIR)/linker/flexpret_hwconfig.ld
 	rm -rf out
+	rm -rf $(CLIENT_BUILD_DIR)
+
+	# If the hwconfig.mk file does not exist, the clean target will fail because
+	# hwconfig.mk is included in the programs/tests makefile. Therefore, create an 
+	# empty file for it to delete.
+	echo "" >> ./hwconfig.mk
```

Comment:
The flow here is a bit weird. But basically what I tried to explain is that the line below (`make -C programs/tests clean`) will run the makefile in programs/tests. That makefile includes hwconfig.mk, and if hwconfig.mk does not exist it will fail.